ironfish/src/wallet/wallet.ts

```diff
@@ -973,12 +981,31 @@ export class Accounts {
 
       await this.db.removeAccount(account, tx)
       await this.db.removeHeadHash(account, tx)
+      await this.db.saveAccountIdToCleanup(account.id, tx)
     })
 
+    this.accountIdsToCleanup.push(account.id)
     this.accounts.delete(account.id)
     this.onAccountRemoved.emit(account)
   }
 
+  async cleanupDeletedAccounts(): Promise<void> {
+    setTimeout(() => void this.cleanupDeletedAccounts(), 100000)
```

Comment:
One of these will get queued up every event loop cycle, was that intended?

Reply:
Were not consuming the return value of setTimeout, which means we're creating a bug with a hanging co-routine. What happens here during shut down?